lib/actions/preload.js

```diff
@@ -0,0 +1,423 @@
+/*
+Copyright 2016-2020 Balena
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import * as _ from 'lodash';
+import { getBalenaSdk, getVisuals } from '../utils/lazy';
+import * as dockerUtils from '../utils/docker';
+
+const isCurrent = commit => commit === 'latest' || commit === 'current';
+
+let allDeviceTypes;
+const getDeviceTypes = function() {
+	const Bluebird = require('bluebird');
+	if (allDeviceTypes !== undefined) {
+		return Bluebird.resolve(allDeviceTypes);
+	}
+	const balena = getBalenaSdk();
+	return balena.models.config
+		.getDeviceTypes()
+		.then(deviceTypes => _.sortBy(deviceTypes, 'name'))
+		.tap(dt => {
+			allDeviceTypes = dt;
+		});
+};
+
+const getDeviceTypesWithSameArch = function(deviceTypeSlug) {
+	return getDeviceTypes().then(function(deviceTypes) {
+		const deviceType = _.find(deviceTypes, { slug: deviceTypeSlug });
+		return _(deviceTypes)
+			.filter({ arch: deviceType.arch })
+			.map('slug')
+			.value();
+	});
+};
+
+const getApplicationsWithSuccessfulBuilds = function(deviceType) {
+	const balenaPreload = require('balena-preload');
+	const balena = getBalenaSdk();
+
+	return getDeviceTypesWithSameArch(deviceType).then(deviceTypes =>
+		balena.pine.get({
```

Comment:
@Page- I never predicted that we would be using generic methods in plan JS :)
The issue is that the typings do not default to `AnyObject` when the T isn't defined (like in here, for obvious reasons), in an effort to enforce using type checked method calls.
You can either:
* simply replace this with `balena.models.application.getAll()` and pass in the options
* or in you really need to use `pine.get()`, define the `pineParams` option right above and add a jsdoc comment like `/** @type {BalenaSdk.PineOptionsFor<BalenaSdk.Appliation>} */`.

I think that this reveals the need of changing the typings to also default to AnyObject when the generic argument for the target resource isn't provided.